libnavigation-core/src/main/java/com/mapbox/navigation/core/history/HistoryFiles.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.navigation.core.history
+
+import android.content.Context
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.base.common.logger.model.Message
+import com.mapbox.base.common.logger.model.Tag
+import com.mapbox.navigation.base.options.HistoryRecorderOptions
+import java.io.File
+
+/**
+ * Create a default directory for the history files to be saved.
+ * When a directory has been provided through options, ensure it exists.
+ */
+internal class HistoryFiles(
+    val applicationContext: Context,
+    val logger: Logger
+) {
+
+    fun absolutePath(options: HistoryRecorderOptions): String? {
```

Comment:
```suggestion
internal object HistoryFiles {

    fun absolutePath(applicationContext: Context, options: HistoryRecorderOptions, logger: Logger): String? {
```
This will never need to have multiple instances, let's make it a singleton.